lib/platform/gitlab/index.ts

```diff
@@ -339,18 +341,18 @@ export async function getBranchStatus(
   logger.debug(`Got res with ${res.length} results`);
   if (res.length === 0) {
     // Return 'pending' if we have no status checks
-    return BRANCH_STATUS_PENDING;
+    return BranchStatus.yellow;
   }
-  let status: BranchStatus = BRANCH_STATUS_SUCCESS;
+  let status: BranchStatus = BranchStatus.green;
   // Return 'success' if all are success
   res.forEach(check => {
     // If one is failed then don't overwrite that
-    if (status !== 'failure') {
+    if (status !== BranchStatus.red) {
```

Comment:
I think we should first map all states. filter (ignore) `check.allow_failure`

Then first check for some failed state and return `red` on failed. 

Second check for some not success and return `yellow` if match. otherwise we can return `green`.

Reply:
what if all of them allow_failure? return yellow or green?

Reply:
I think `green` because there is no required status